clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -467,37 +467,48 @@ else if (strategy == OffsetResetStrategy.LATEST)
      *         the defaultResetPolicy is NONE
      */
     public Map<TopicPartition, List<ConsumerRecord<K, V>>> fetchedRecords() {
+        if (nextInLineException != null) {
```

Comment:
Good point. I have updated the patch to throw exception only if the partition is fetchable and the subscription position equals the fetched offset of the record that causes the exception.